presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveDynamicPartitionPruning.java

```diff
@@ -0,0 +1,241 @@
+        // Probe-side is fully scanned
+        assertEquals(probeStats.getInputPositions(), countRows("lineitem"));
+        assertEquals(probeStats.getDynamicFilterSplitsProcessed(), 0L);
+
+        DynamicFiltersStats dynamicFiltersStats = getDynamicFilteringStats(result.getQueryId());
+        assertEquals(dynamicFiltersStats.getTotalDynamicFilters(), 1L);
+        assertEquals(dynamicFiltersStats.getLazyDynamicFilters(), 1L);
+        assertEquals(dynamicFiltersStats.getReplicatedDynamicFilters(), 0L);
+        assertEquals(dynamicFiltersStats.getDynamicFiltersCompleted(), 1L);
+
+        DynamicFilterDomainStats domainStats = getOnlyElement(dynamicFiltersStats.getDynamicFilterDomainStats());
+        assertEquals(domainStats.getSimplifiedDomain(), Domain.create(ValueSet.ofRanges(
+                range(INTEGER, 1L, true, 100L, true)), false)
+                .toString(getSession().toConnectorSession()));
+        assertEquals(domainStats.getDiscreteValuesCount(), 0);
+        assertEquals(domainStats.getRangeCount(), 1);
+    }
+
+    @Test
+    public void testJoinWithMultipleDynamicFiltersOnProbe()
+    {
+        DistributedQueryRunner runner = (DistributedQueryRunner) getQueryRunner();
+        // supplier names Supplier#000000001 and Supplier#000000002 match suppkey 1 and 2
+        ResultWithQueryId<MaterializedResult> result = runner.executeWithQueryId(
+                getSession(),
+                "SELECT * FROM (" +
+                        "SELECT supplier.suppkey FROM " +
+                        "lineitem JOIN tpch.tiny.supplier ON lineitem.suppkey = supplier.suppkey AND supplier.name IN ('Supplier#000000001', 'Supplier#000000002')" +
+                        ") t JOIN tpch.tiny.partsupp ON t.suppkey = partsupp.suppkey AND partsupp.suppkey IN (2, 3)");
+        assertGreaterThan(result.getResult().getRowCount(), 0);
+
+        OperatorStats probeStats = searchScanFilterAndProjectOperatorStats(result.getQueryId(), TPCH_PARTITIONED_SCHEMA + ":lineitem");
+        // Probe-side is partially scanned
+        assertLessThan(probeStats.getInputPositions(), countRows("lineitem"));
+        assertEquals(probeStats.getDynamicFilterSplitsProcessed(), 0L);
+
+        DynamicFiltersStats dynamicFiltersStats = getDynamicFilteringStats(result.getQueryId());
+        assertEquals(dynamicFiltersStats.getTotalDynamicFilters(), 2L);
+        assertEquals(dynamicFiltersStats.getLazyDynamicFilters(), 2L);
+        assertEquals(dynamicFiltersStats.getReplicatedDynamicFilters(), 0L);
+        assertEquals(dynamicFiltersStats.getDynamicFiltersCompleted(), 2L);
+
+        List<DynamicFilterDomainStats> domainStats = dynamicFiltersStats.getDynamicFilterDomainStats();
+        assertEquals(domainStats.size(), 2L);
+        domainStats.forEach(stats -> {
+            assertGreaterThanOrEqual(stats.getRangeCount(), 1);
+            assertEquals(stats.getDiscreteValuesCount(), 0);
```

Comment:
could we add a test for `getDiscreteValuesCount` too? I think you can use `row_type` for this (I don't think row types are orderable).

Reply:
RowType is orderable as long as the fields within are themselves orderable. I think we can use ColorType. 
For testing getDiscreteValuesCount specifically, would it be ok we add a test in TestDynamicFilterService instead (maybe separate PR) ? It will be much easier to do it there.

Reply:
> would it be ok we add a test in TestDynamicFilterService instead (maybe separate PR)

Yes, that works too